### DISCUSSION TITLE
Please any ask any questions you may have about https://trustfree.market

### DISCUSSION THREAD

**Discussion by firepacket** [firepacket/Trustfreemarket#5] [General]
The site is fully operational and pays out your escrow as a paper wallet (we will have onsite send functionality soon)

Please ask any questions here. I hope we can have a safe, free, bitcoin marketplace!

**Comment by darkarmystudios** [comment]
Hi and thanks for the effort of putting up this marketplace. Just wanted to know what database to use with it ? Can it work with MongoDB or MariaDB ?

**Reply by firepacket** [reply]
It's built to be immediately operable with MSSQL or MSSQL Express, but you should be able to use any relational SQL database.

**Reply by firepacket** [reply]
My apologies it's been a while but it looks like someone has written a driver that can interface MongoDB with EntityFramework: https://www.cdata.com/kb/tech/mongodb-ado-codefirst.rst
You could also try: https://www.nuget.org/packages/MongoDB.EntityFramework

**Reply by firepacket** [reply]
But be aware I designed the project DB first, not Code first, so you might have some work to do converting all the objects.

**Reply by darkarmystudios** [reply]
Thanks for the quick response. I will try to run it with mssql or mariadb first. And then slowly start converting to a nosql format for the objects.

**Reply by darkarmystudios** [reply]
One more question. Will this work straight out the box with MariaDB/MySQL ?

**Reply by firepacket** [reply]
Not quite out of the box, but it *might* work with mysql with a few config changes, maybe a single package install. MariaDB will probably need some extra code.

**Reply by firepacket** [reply]
This is an old project and honestly I don't always agree with the direction technology moves, people care very little about backward compatibility nowadays.

**Reply by darkarmystudios** [reply]
I agree. But this is amazing work. probably the only of its kind so far that I know of.

**Reply by firepacket** [reply]
Thank you. I feel like it's kind of lost technology that is maybe being overlooked due to other transactional methods, but this method has advantages that none of the other solutions have. I hope we will be able to bring something to market soon on our main active site.

There is a 2nd component which I haven't uploaded which passively watches the blockchain for incoming transactions to our escrow address and allows for instant feedback to the site through Named Pipes when a transaction is seen, unfortunately It doesn't see transactions coming from new Bech32 addresses so I have to use a timer to check a 3rd party site for the balance until I can figure out a way to do it properly myself. I am looking for a way to monitor random addresses without outgoing queries so once I figure this out, I think we will be able to launch. I have a lot of other code that hasn't been merged yet, namely the secure encrypted chat in the escrow view, and payment QR codes with realtime feedback, which will speed up transactions dramatically.
